From: Jan <cloux@rote.ch>
Date: Fri, 8 Mar 2019 21:18:35 +0100
Subject: fix: resource leaks and other issues as reported by infer

 to run infer static code analyzer (fbinfer.com):
 cd src; infer run -- make

 - fix: fd_move.c resource leak - close file handle
 - fix: runit-init.c resource leak - close file handle
 
---

 src/fd_move.c       |  5 ++++-
 src/runit-init.c    |  6 +++---
 2 files changed, 7 insertions(+), 4 deletions(-)

--- a/src/fd_move.c
+++ b/src/fd_move.c
@@ -6,7 +6,10 @@
 int fd_move(int to,int from)
 {
   if (to == from) return 0;
-  if (fd_copy(to,from) == -1) return -1;
+  if (fd_copy(to,from) == -1) {
+    close(from);
+    return -1;
+  }
   close(from);
   return 0;
 }
--- a/src/runit-init.c
+++ b/src/runit-init.c
@@ -17,7 +17,7 @@
 void usage(void) { strerr_die4x(0, "usage: ", progname, USAGE, "\n"); }
 
 void runit_halt () {
-  if (open_trunc(STOPIT) == -1)
+  if (close(open_trunc(STOPIT)) == -1)
     strerr_die4sys(111, FATAL, "unable to create ", STOPIT, ": ");
   if (chmod(STOPIT, 0100) == -1)
     strerr_die4sys(111, FATAL, "unable to chmod ", STOPIT, ": ");
@@ -29,11 +29,11 @@
 }
 
 void runit_reboot () {
-  if (open_trunc(STOPIT) == -1)
+  if (close(open_trunc(STOPIT)) == -1)
     strerr_die4sys(111, FATAL, "unable to create ", STOPIT, ": ");
   if (chmod(STOPIT, 0100) == -1)
     strerr_die4sys(111, FATAL, "unable to chmod ", STOPIT, ": ");
-  if (open_trunc(REBOOT) == -1)
+  if (close(open_trunc(REBOOT)) == -1)
     strerr_die4sys(111, FATAL, "unable to create ", REBOOT, ": ");
   if (chmod(REBOOT, 0100) == -1)
     strerr_die4sys(111, FATAL, "unable to chmod ", REBOOT, ": ");
